standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -3799,10 +3799,33 @@ public Table get_table_core(GetTableRequest getTableRequest) throws MetaExceptio
   @Override
   public GetTablesResult get_table_objects_by_name_req(GetTablesRequest req) throws TException {
     String catName = req.isSetCatName() ? req.getCatName() : getDefaultCatalog(conf);
+    if (isDatabaseRemote(req.getDbName())) {
+      return new GetTablesResult(getRemoteTableObjectsInternal(req.getDbName(), req.getTblNames(), req.getTablesPattern()));
+    }
     return new GetTablesResult(getTableObjectsInternal(catName, req.getDbName(),
         req.getTblNames(), req.getCapabilities(), req.getProjectionSpec(), req.getTablesPattern()));
   }
 
+  private String tableNames2regex(List<String> tableNames) {
+    return "/^(" + String.join("|", tableNames) + ")$/";
```

Comment:
Are there any nuances around how different RDBMS database drivers process this regex? Do we know that this is truely a regex or do certain drivers expect other things?

Reply:
Good catch. Will look into that. As a side note, it seems that at least mysql does not like ".*" to be passed in as a regex getTable API call and will throw an error. I have to manually changed the regex to null, is it normal?

Reply:
After some digging around. It seems that tableNamePattern in DatabaseMetaData.getTables() is a bit different from regular regex we are using. In order to search for all tables, the regex should be "%" or null instead of ".*". And we cannot use tableNamePattern to match multiple tableNames([reference](url)). I have created another helper function to get around this issue, and it is included in the latest commit!